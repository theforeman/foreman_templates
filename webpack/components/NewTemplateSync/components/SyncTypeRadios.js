import React from 'react';

import { Radio } from 'patternfly-react';
import PropTypes from 'prop-types';

import CommonForm from 'foremanReact/components/common/forms/CommonForm';

const SyncTypeRadios = ({
  controlLabel,
  radios,
  name,
  className = '',
  inputClassName = 'col-md-6',
  disabled = false,
}) => (
  <CommonForm
    label={controlLabel}
    className={className}
    inputClassName={inputClassName}
  >
    {radios.map((item, index) => (
      <Radio
        key={index}
        inline
        title={item.label}
        checked={item.checked}
        disabled={disabled}
        value={item.value}
        onChange={item.onChange}
      >
        {item.label}
      </Radio>
    ))}
  </CommonForm>
);

SyncTypeRadios.propTypes = {
  controlLabel: PropTypes.string.isRequired,
  radios: PropTypes.array.isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  disabled: PropTypes.bool,
};

SyncTypeRadios.defaultProps = {
  className: undefined,
  inputClassName: undefined,
  disabled: false,
};

export default SyncTypeRadios;
