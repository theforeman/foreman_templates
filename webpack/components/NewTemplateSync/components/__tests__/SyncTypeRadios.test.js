import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import SyncTypeRadios from '../SyncTypeRadios';

const noop = () => {};

const fixtures = {
  'should render': {
    controlLabel: 'Test radios',
    name: 'radios',
    radios: [
      { label: 'A', checked: false, value: 'a', onChange: noop },
      { label: 'B', checked: true, value: 'b', onChange: noop },
      { label: 'C', checked: false, value: 'c', onChange: noop },
    ],
  },
};

describe('SyncTypeRadios', () =>
  testComponentSnapshotsWithFixtures(SyncTypeRadios, fixtures));
