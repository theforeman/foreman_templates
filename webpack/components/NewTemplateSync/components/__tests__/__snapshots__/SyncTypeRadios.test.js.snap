// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncTypeRadios should render 1`] = `
<CommonForm
  className=""
  inputClassName="col-md-6"
  label="Test radios"
>
  <Radio
    bsClass="radio"
    checked={false}
    disabled={false}
    inline={true}
    key="0"
    onChange={[Function]}
    title="A"
    value="a"
  >
    A
  </Radio>
  <Radio
    bsClass="radio"
    checked={true}
    disabled={false}
    inline={true}
    key="1"
    onChange={[Function]}
    title="B"
    value="b"
  >
    B
  </Radio>
  <Radio
    bsClass="radio"
    checked={false}
    disabled={false}
    inline={true}
    key="2"
    onChange={[Function]}
    title="C"
    value="c"
  >
    C
  </Radio>
</CommonForm>
`;
